src/test/java/org/logstash/beats/ServerTest.java

```diff
@@ -34,9 +42,43 @@ public void setUp() {
     @Test
     public void testServerShouldTerminateConnectionIdleForTooLong() throws InterruptedException {
         int inactivityTime = 3; // in seconds
+        int concurrentConnections = 10;
+
+        final Random random = new Random();
+
+        final CountDownLatch latch = new CountDownLatch(concurrentConnections);
 
         final Server server = new Server(host, randomPort, inactivityTime);
+        final AtomicBoolean otherCause = new AtomicBoolean(false);
+        server.setMessageListener(new MessageListener() {
+            public void onNewConnection(ChannelHandlerContext ctx) {
+                // Make sure connection is closed on exception too.
+                if (random.nextInt(10) < 1) {
```

Comment:
I'm a little confused, we are connecting 10 clients, with a 10% chance of throwing an exception (which we are expecting), isn't this pretty likely not to fire?